NoahPBM/contracts/paymentmanager/NoahPaymentManager.sol

```
@@ -251,24 +251,18 @@ contract NoahPaymentManager is Ownable, Pausable, AccessControl, INoahPaymentSta
         require(Address.isContract(erc20Token), "Must be a valid ERC20 smart contract");
         require(erc20TokenValue > 0, "Token value should be more than 0");
         require(bytes(paymentUniqueId).length != 0);
+        require(
+            pendingPBMTokenBalance[campaignPBM][erc20Token] >= erc20TokenValue,
```

noted, now it wont fail silently due to if condition not executing.